services/QuillLMS/app/models/bonus_days_calculator.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class BonusDaysCalculator < ApplicationService
+  JUNE = 6
+  JULY = 7
+  DECEMBER = 12
+
+  attr_reader :user
+
+  delegate :school, to: :user
+
+  def initialize(user)
```

Comment:
I think you have a really good testing setup, so you can ignore this, but just something to think about in the future: 

Having a `current date/time` within a function can be tricky to test (hence callbacks and time travel), BUT if you can refactor the function to pass the `current date/time` as a parameter, then it becomes a pure function, in that it no longer has the "changing time side effect". 

Pure functions are very easy to test (there are other benefits that don't fully apply here, like being able to run it for the past or future). e.g.

```ruby
attr_reader :user, :today

def initialize(user, today: Date.current)
  @user = user
  @today = today
end

... tests
expect BonusDaysCalculator.run(user, today: Date.new(2003,6,1)).to eq(29)

Reply:
I would agree.  I will just refactor now.

Reply:
I would agree that making this a pure function would be an improvement.  I will refactor now.